src/Reference/HasMany.php

```diff
@@ -29,7 +29,11 @@ public function getTheirFieldName(): string
         // this is pure guess, verify if such field exist, otherwise throw
         // TODO probably remove completely in the future
         $ourModel = $this->getOurModel(null);
-        $theirFieldName = $this->getModelTableString($ourModel) . '_' . $ourModel->id_field;
+        $theirFieldName = $this->getModelTableString($ourModel);
+        if (($dbSepPos = strpos($theirFieldName, '.')) === false) {
+            $dbSepPos = -1;
+        }
+        $theirFieldName = substr($theirFieldName, $dbSepPos + 1) . '_' . $ourModel->id_field;
```

Comment:
```suggestion
        $theirFieldName = preg_replace('~^.+?\.~', '', $this->getModelTableString($ourModel)) . '_' . $ourModel->id_field;
```

Reply:
is such guessing present solely for HasMany?

Reply:
I keep reminding myself to get better at regex.....and I was so focused on eliminating the exception that I never noticed getTheirFieldName being public so that I could use it in the tests.

Both items have been addressed.  I did check, and only found that issue in HasMany -- it wouldn't apply to HasOne, which specifies our field name directly and -- unless you say otherwise -- uses the Id field of the other Model, so there's no guessing involved.  The comments in getTheirFieldName seem to indicate that someone doesn't like the guessing at all, and that it may eventually go away.  From my point of view, though, well-documented standards (i.e., using 'user_id' as the default field name for the id of a record in the User model) that make people's lives easier are a good thing, as long as they don't inhibit functionality (which this doesn't, since you can always override the field if you want to).

The pull request has been updated and passes all tests (the smoke test issues in other code remains, but that's from items not in this pull request).